docs/management-sdk-migration.md

```diff
@@ -0,0 +1,115 @@
+# `ServiceManagerBuilder` to `ServiceHubContextBuilder` Migration guidance
+
+This guidance is intended to assist in the migration to `ServiceHubContextBuilder` from `ServiceManagerBuilder`. It will focus on side-by-side comparisons for similar operations between the legacy and the new APIs. The new APIs will be released in version 10.0.0 .
+
+We assume that you are familiar with the usage of `ServiceManagerBuilder` and the related APIs, otherwise, please refer the [Azure SignalR Service Management SDK ](management-sdk-guide.md) other than this guidance.
+- [`ServiceManagerBuilder` to `ServiceHubContextBuilder` Migration guidance](#servicemanagerbuilder-to-servicehubcontextbuilder-migration-guidance)
+  - [Migration Benefits](#migration-benefits)
+  - [Change Overview](#change-overview)
+  - [API comparisons](#api-comparisons)
+    - [Entry point](#entry-point)
+    - [Negotiate](#negotiate)
+    - [Send Messages and Manage Groups](#send-messages-and-manage-groups)
+    - [Health check](#health-check)
+
+
+
+## Migration Benefits
+* The new APIs provide more functionalities to manage your clients and groups, such as closing a connection by connection id, checking if a connection exists, if a user exists, if a group exists. 
+* The new APIs provide more options for negotiation, such as whether the client is a diagnostic client.
+* The new APIs are more friendly for negotiation with multiple SignalR Service instances. `IServiceManager.GetClientEndpoint` and `IServiceManager.GenerateClientAccessToken` are combined into one method to make sure the client endpoint and the access token come from the same SignalR Service endpoint. An `HttpContext` instance is allowed to passed into the endpoint router to provide more information for the routing. <!--Todo Add link about sharding doc-->
```

Comment:
to passed -> to pass

Reply:
`<!--Todo Add link about sharding doc-->` also include in this PR?

Reply:
Sharding doc is not ready. It has to wait for this doc.